syntax = "proto3";

package complex_message;

message SimpleMessage {
  int64 message_id = 1;
  MessageType type = 2;
  bytes content = 3;
  repeated int32 flags = 4;

  enum MessageType {
    UNKNOWN = 0;
    REQUEST = 1;
    RESPONSE = 2;
    NOTIFICATION = 3;
    ACK = 4;
  }
}

message ResponseMessage {
  int64 original_message_id = 1;
  bytes response_content = 2;
  ResponseStatus status = 3;

  enum ResponseStatus {
    OK = 0;
    ERROR = 1;
  }
}